src/Agent.Listener/Configuration/AutoLogonRegistryManager.cs

```diff
@@ -0,0 +1,339 @@
+            RevertOriginalValue(hive, $"{securityId}\\{RegistryConstants.SubKeys.ScreenSaverDomainPolicy}", RegistryConstants.ValueNames.ScreenSaver);
+            RevertOriginalValue(hive, $"{securityId}\\{RegistryConstants.SubKeys.StartupProcess}", RegistryConstants.ValueNames.StartupProcess);
+        }
+
+        private void SetValue(RegistryHive targetHive, string subKeyName, string name, string value)
+        {
+            //take backup if it exists
+            string origValue = _registryManager.GetValue(targetHive, subKeyName, name);
+            if (!string.IsNullOrEmpty(origValue))
+            {
+                var nameForTheBackupValue = GetBackupValueName(name);
+                _registryManager.SetValue(targetHive, subKeyName, nameForTheBackupValue, origValue);
+            }
+
+            _registryManager.SetValue(targetHive, subKeyName, name, value);
+        }
+
+        private string GetBackupValueName(string valueName)
+        {
+            return string.Concat(RegistryConstants.BackupKeyPrefix, valueName);
+        }
+
+        private void ShowAutoLogonWarningIfAlreadyEnabled(string domainName, string userName)
+        {
+            //we cant use store here as store is specific to the agent root and if there is some other on the agent, we dont have access to it
+            //we need to rely on the registry only
+            FetchAutoLogonUserDetails(out string autoLogonUserName, out string autoLogonUserDomainName);
+            
+            if (autoLogonUserName != null
+                    && autoLogonUserDomainName != null
+                    && !domainName.Equals(autoLogonUserDomainName, StringComparison.CurrentCultureIgnoreCase)
+                    && !userName.Equals(autoLogonUserName, StringComparison.CurrentCultureIgnoreCase))
+            {
+                var terminal = HostContext.GetService<ITerminal>();
+                terminal.WriteLine(StringUtil.Loc("AutoLogonAlreadyEnabledWarning", userName));
+            }
+        }
+
+        private void FetchAutoLogonUserDetails(out string userName, out string domainName)
+        {
+            userName = null;
+            domainName = null;
+
+            var regValue = _registryManager.GetValue(RegistryHive.LocalMachine, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogon);
+            if (int.TryParse(regValue, out int autoLogonEnabled)
+                    && autoLogonEnabled == 1)
+            {
+                userName = _registryManager.GetValue(RegistryHive.LocalMachine, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogonUserName);
+                domainName = _registryManager.GetValue(RegistryHive.LocalMachine, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogonDomainName);
+            }
+        }
+    }
+
+    public class RegistryConstants
```

Comment:
it would be helpful for readers if the subkeys/valuenames were organized based on their classification. either by splitting into separate classes or at least with a comment and a line of whitespace between the two groups. half are machine (autologon) settings, and half are user (startup/etc) settings. i feel like it would help the reader understand the config flow.